sdk/src/main/java/com/mapbox/maps/renderer/widget/WidgetPosition.kt

```diff
@@ -1,17 +1,112 @@
 package com.mapbox.maps.renderer.widget
 
 import com.mapbox.maps.MapboxExperimental
+import kotlin.math.abs
 
 /**
  * Specifies widget position relative to the screen.
- * @param horizontal position relative to the screen.
- * @param vertical position relative to screen.
+ *
+ * @property horizontalAlignment the horizontal position relative to the screen.
+ * @property verticalAlignment the vertical position relative to screen.
+ * @property offsetX the horizontal offset in pixels towards the right of the screen.
+ * @property offsetY the vertical offset in pixels towards the right of the screen.
  */
 @MapboxExperimental
-class WidgetPosition(
-  val horizontal: Horizontal,
-  val vertical: Vertical,
+class WidgetPosition private constructor(
+  val horizontalAlignment: Horizontal,
+  val verticalAlignment: Vertical,
+  val offsetX: Float,
+  val offsetY: Float
 ) {
+  @Deprecated(
+    message = "Direct constructor of WidgetPosition is deprecated, and might be removed in future releases.",
+    replaceWith = ReplaceWith("WidgetPosition.Builder()")
+  )
+  constructor(horizontal: Horizontal, vertical: Vertical) : this(horizontal, vertical, 0f, 0f)
+
+  /**
+   * Returns a String for the object.
+   */
+  override fun toString() =
+    "WidgetPosition(horizontalAlignment=$horizontalAlignment,verticalAlignment=$verticalAlignment,offsetX=$offsetX,offsetY=$offsetY"
```

Comment:
```suggestion
    "WidgetPosition(horizontalAlignment=$horizontalAlignment,verticalAlignment=$verticalAlignment,offsetX=$offsetX,offsetY=$offsetY")
```

Reply:
this is fixed already, sorry have pushed too many commits lately.